spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/StaticGraph.scala

```diff
@@ -147,4 +153,59 @@ class StaticGraph[T: ClassTag](
     gradInput = fetchModelGradInput()
     gradInput
   }
+
+  private var inputFormats: Seq[Int] = null
+  private var outputFormats: Seq[Int] = null
+
+  /**
+   * set input formats for graph
+   * @param formats
+   * @return
+   */
+  def setInputFormats(formats: Seq[Int]): this.type = {
+    inputFormats = formats
+    this
+  }
+
+  /**
+   * set output formats for graph
+   * @param formats
+   * @return
+   */
+  def setOutputFormats(formats: Seq[Int]): this.type = {
+    outputFormats = formats
+    this
+  }
+
+  /**
+   * convert static graph to ir graph and build according to engine type
+   * @return
+   */
+  def toIRgraph() : IRGraph[T] = {
+    val inFormats = if (inputFormats == null) {
+      logger.warn(s"Please set input formats for graph model, otherwise we" +
+        s"will use default input format NCHW")
+      Seq(Memory.Format.nchw)
+    } else inputFormats
+
+    val outFormats = if (outputFormats == null) {
+      logger.warn(s"Please set output formats for graph model, otherwise we" +
+        s"will use default output format NC")
+      Seq(Memory.Format.nc)
+    } else outputFormats
+
+    val allNodes = forwardExecution
+    if (!BlasToIR[T].convertingCheck(allNodes)) return null
+    inFormats.foreach(in => if (in == Memory.Format.nhwc) return null)
```

Comment:
print a warning message saying we cannot support nhwc for now ?

Reply:
ok, changed.